lib/object_store.rb

```diff
@@ -0,0 +1,39 @@
+module ObjectStore
+  class UnknownItemType < StandardError
+    def initialize(item_type)
+      super("Unknown item type: #{item_type}")
+    end
+  end
+
+  def self.configuration
+    @configuration ||= OpenStruct.new
+  end
+
+  def self.configure
+    yield(configuration)
+  end
+
+  def self.config_for_item_type(item_type)
+    configuration.fields[item_type] || raise(UnknownItemType, item_type)
+  end
+
+  def self.fields_for_item_type(item_type)
```

Comment:
It feels like there could be two small objects here: `item_type` and `field` as a separate concept from "field_definition" (the config representation).

We already want to send messages to them such as `item_type.required?` and `field.required?`, and I imagine this behaviour is going to extend somewhat?

Reply:
That's a good call that, still getting my head back into a Ruby-ish way of thinking!